components/higher-order/with-dragging/index.js

```diff
@@ -0,0 +1,163 @@
+/**
+ * WordPress Dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal Dependencies
+ */
+import './style.scss';
+
+const withDragging = ( OriginalComponent ) => {
+	class Draggable extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.onDragStart = this.onDragStart.bind( this );
+			this.onDragOver = this.onDragOver.bind( this );
+			this.onDragEnd = this.onDragEnd.bind( this );
+			this.cursorTop = null;
+			this.cursorLeft = null;
+			this.config = {
+				dragImageClass: 'components-with-dragging__invisible-drag-image',
+				cloneWrapperClass: 'components-with-dragging__clone',
+				cloneHeightTransformationBreakpoint: 700,
+				clonePadding: 20,
+			};
+			this.state = {
+				cloneNodeId: null,
+				elementId: null,
+			};
+		}
+
+		/**
+		 * Removes the element clone, resets cursor, and removes drag listener.
+		 * @param  {Object} event     The non-custom DragEvent.
+		 */
+		onDragEnd( event ) {
+			const { elementId, cloneNodeId } = this.state;
+			const element = document.getElementById( elementId );
+			const cloneWrapper = document.getElementById( cloneNodeId );
+
+			if ( element && cloneWrapper ) {
+				// Remove clone.
+				element.parentNode.removeChild( cloneWrapper );
+			}
+
+			// Reset cursor.
+			document.body.classList.remove( 'dragging' );
+
+			this.setState( { elementId: null, cloneNodeId: null } );
+
+			document.removeEventListener( 'dragover', this.onDragOver );
+			event.stopPropagation();
+		}
+
+		/*
+		 * Updates positioning of element clone based on mouse movement during dragging.
+		 * @param  {Object} event     The non-custom DragEvent.
+		 */
+		onDragOver( event ) {
+			const { cloneNodeId } = this.state;
+			const cloneWrapper = document.getElementById( cloneNodeId );
+
+			cloneWrapper.style.top =
+				`${ parseInt( cloneWrapper.style.top, 10 ) + parseInt( event.clientY, 10 ) - parseInt( this.cursorTop, 10 ) }px`;
```

Comment:
Are `event.clientY` and `this.cursorTop` not expected to be numeric? Do we need the `parseInt`?

Reply:
If I remember correctly there was some issue with these where some browser was specifying some of these as strings with `px` in the end. Parsing to Int was needed to get the number out of it. I need to investigate this, cause it was an issue I had at the very early stages.

Reply:
I might expect this with `style.top`, or values returned by `window.getComputedStyle`, as these would represent the assigned style value (including unit), but `clientY` by specification should be guaranteed as numeric, and parsing might imply otherwise that we'd not expect it to be.

https://www.w3.org/TR/uievents/#interface-mouseevent

Reply:
You are absolutely right, I think it was `style.top`. I will confirm and update these.